src/autorest-core/lib/data-store/data-store.ts

```diff
@@ -320,8 +321,28 @@ export class DataStore extends DataStoreView {
     return Object.getOwnPropertyNames(this.store);
   }
 
-  public Blame(absoluteUri: string, position: SmartPosition): BlameTree {
+  public Blame(absoluteUri: string, position: SmartPosition, message?: (m: Message) => void): BlameTree {
     const data = this.ReadStrictSync(absoluteUri);
+
+    // verify if the path exists in the document
+    if ('path' in position) {
+      let x = (<any>JSON.parse(data.ReadData()));
+      let keys = <string[]>(<any>position)['path'];
+      let validKeys = [];
+      let currObj = x;
+      for (let i = 0; i < keys.length; ++i) {
+        if (!(keys[i] in currObj)) {
+          if (message !== undefined) {
+            message({ Channel: Channel.Warning, Text: 'Could not find exact path: \'' + keys + '\'', Details: position });
+          }
+          break;
+        }
+        validKeys.push(keys[i]);
+        currObj = currObj[keys[i]];
+      }
+      (<any>position)['path'] = validKeys;
+    }
+
     const resolvedPosition = CompilePosition(position, data);
```

Comment:
verbose or not, your code looks like it would do what it should! 👍 
Howeeeeeever I'd suggest the following approach: just call out to `CompilePosition` and see whether it succeeds, if not work your way up and retry with shorter paths (and only if `message` is provided - maybe call that `onPathLookupError` to indicate that providing it means errors are no longer thrown).

This may sound less efficient, but actually it is the opposite! Here is why:
Objects we encounter here can be HUGE as in ~30k lines when JSON serialized (think fully resolved network spec meaning that even all samples are inlined!). Parsing those is noticably expensive and with your approach you do that for *EVERY* `Blame` (there are easily thousands of calls e.g. when building full source maps, as in `.map` files).
Okay, one could do smart caching (`ReadObject` does something close to caching, not that the object it returns is supposed to be mutable without changing the cached copy...) but we can do even better: the data store caches YAML ASTs of objects (those are considered immutable) and *that's* what `CompilePosition` operates on, making it faster than anything that involves parsing objects and such 😃 

Furthermore, we expect that the 99% case is that `CompilePosition` simply succeeds! So we can achieve 0 overhead for the 99% case using the "backing off on error" strategy